netbox/dcim/forms/model_forms.py

```diff
@@ -1435,6 +1524,21 @@ class Meta:
             'mode': '802.1Q Mode',
         }
 
+    def clean_mac_address(self):
```

Comment:
We can't effect the creation of related objects under field validation, because it could result in an inconsistent state should later validations fail. This needs to be handled entirely under `save()`.